services/cd-service/pkg/repository/repository.go

```diff
@@ -1122,6 +1132,18 @@ func (r *repository) Apply(ctx context.Context, transformers ...Transformer) err
 			}
 		}
 		r.notify.Notify()
+
+		var changedAppNames []string
+		var seen = make(map[string]bool)
+		for _, app := range changes.ChangedApps {
+			if _, ok := seen[app.App]; !ok {
+				seen[app.App] = true
+				changedAppNames = append(changedAppNames, app.App)
+			}
+		}
+		if len(changedAppNames) != 0 {
+			r.notify.NotifyChangedApps(changedAppNames)
+		}
```

Comment:
this code looks the same as in ProcessQueueOnce. Please extract this into a function.